examples/vision/mlpmixer_image_classification.py

```diff
@@ -0,0 +1,275 @@
+"""
+Title: Image Classification with MLP-Mixer
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/05/30
+Last modified: 2021/05/30
+Description: Implementing the MLP-Mixer model for image classification.
+"""
+
+"""
+## Introduction
+
+This example implements the [MLP-Mixer](https://arxiv.org/abs/2105.01601)
+model, by Ilya Tolstikhin et al., for image classification, and demonstrates it
+on the CIFAR-100 dataset. The MLP-Mixer is an architecture based exclusively on
+multi-layer perceptrons (MLPs), and contains two types of MLP layers:
+
+1. One applied independently to image patches, which mixes the per-location features.
+2. The other applied across patches (across channels), which mixes spatial information.
```

Comment:
1 + 2 is what a depthwise separable convolution layer does (sepconv has quadratically fewer parameters than conv). Mathematically this architecture is a depthwise separable CNN, not unlike Xception, but with 2 dense transforms chained instead of just one (also with max pooling removed and BN replaced with layernorm).

Hopefully this will be raised during peer review, but probably not.

Worth pointing out this context in the description.

Reply:
+1